dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/sms/EnrollmentSMSListener.java

```diff
@@ -185,23 +190,27 @@ protected SmsResponse postProcess(IncomingSms sms, SmsSubmission submission, Use
     // TODO: Unsure about this handling for enrollments, this needs to be
     // checked closely
     Enrollment enrollment;
-    boolean enrollmentExists = enrollmentService.enrollmentExists(enrollmentid.getUid());
+    boolean enrollmentExists = apiEnrollmentService.enrollmentExists(enrollmentid.getUid());
     if (enrollmentExists) {
-      enrollment = enrollmentService.getEnrollment(enrollmentid.getUid());
+      try {
+        enrollment = enrollmentService.getEnrollment(enrollmentid.getUid());
```

Comment:
You need to pass the user in. This does not have the user set in the context like in an HTTP request context.